client/lib/DomExtender.ts

```diff
@@ -155,6 +161,19 @@ const NodeExtensionFns: INodeExtensionFns = {
     );
     return createSuperNode(newPath, awaitedOptions);
   },
+  async $extract<T = any>(extractFn: IExtractElementFn<T>, options: IExtractElementOptions = {}): Promise<T> {
```

Comment:
This is clever. I like it! Took me a second. Mix of what we talked about and an actual "reason" to extract it now.